src/VisualStudio/Core/Def/Implementation/ChangeSignature/ChangeSignatureDialogViewModel.cs

```diff
@@ -356,10 +371,66 @@ public bool IsOkButtonEnabled
 
                 _selectedIndex = newSelectedIndex;
 
-                NotifyPropertyChanged("CanMoveUp");
-                NotifyPropertyChanged("CanMoveDown");
-                NotifyPropertyChanged("CanRemove");
-                NotifyPropertyChanged("CanRestore");
+                NotifyPropertyChanged(nameof(CanMoveUp));
+                NotifyPropertyChanged(nameof(MoveUpAutomationText));
+                NotifyPropertyChanged(nameof(CanMoveDown));
+                NotifyPropertyChanged(nameof(MoveDownAutomationText));
+                NotifyPropertyChanged(nameof(CanRemove));
+                NotifyPropertyChanged(nameof(RemoveAutomationText));
+                NotifyPropertyChanged(nameof(CanRestore));
+                NotifyPropertyChanged(nameof(RestoreAutomationText));
+            }
+        }
+
+        public string MoveUpAutomationText
+        {
+            get
+            {
+                if (!CanMoveUp)
+                {
+                    return string.Empty;
```

Comment:
It's a good point, and I'm not sure what the StateOfTheArt is for this problem, but this is how all the UIs in VS work and it's certainly the approach that WPF pushes us toward. When the element is disabled, it doesn't receive focus so it's never read anyway (I think I could put any text here I want, and it'd never get read).